Proxy/faraday_uart_stack/layer_4_service.py

```diff
@@ -139,8 +139,13 @@ def RxPortHasItem(self, service_number):
         try:
             return not self.receive_parsed_queue_dict[service_number].empty()
         except:
-            self.receive_service_queue_open(service_number, self.QUEUE_SIZE_DEFAULT)
-            pass
+            return None
+
+    def RxPortListOpen(self):
+        return self.receive_parsed_queue_dict.keys()
```

Comment:
OK so when proxy performs `RXPortListOpen()` within that function it ends up running `receive_service_queue_open()` if an exception occurs. Cool.